packages/backend/app/Services/Telegram.ts

```diff
@@ -1,54 +1,53 @@
-import axios from 'axios'
-import telegramConfiguration from 'Config/telegram'
-import { DateTime } from 'luxon'
+import { ItelegramConfiguration } from './../../common/interfaces/Itelegram';
+import axios from "axios";
+import telegramConfiguration from "Config/telegram";
+import { DateTime } from "luxon";
+
 
 class TelegramBot {
-  private readonly config: typeof telegramConfiguration
-  constructor () {
-    this.config = telegramConfiguration
+  private readonly config: ItelegramConfiguration;
+  constructor() {
+    this.config = telegramConfiguration;
   }
 
   private async sendRequest<T extends Record<string, void>>(
     path: string,
     _data?: Record<string, void>
   ): Promise<{ ok: boolean } & Partial<T>> {
     const response = await axios.get(
-      `https://api.telegram.org/bot${this.config.test_token}${path}`
-    )
-    return response.data
+      `https://api.telegram.org/bot${this.config.telegram_bot_token}${path}`
+    );
+    return response.data;
   }
 
-  public async sendMsgToTelegram (msg: string): Promise<boolean> {
+  public async sendMsgToTelegram(msg: string): Promise<boolean> {
     return (
-      (
-        await this.sendRequest(
-          `/sendMessage?&chat_id=${this.config.test_chat_id}&parse_mode=html&text=${msg}`
-        )
-      ).ok
-    )
+      await this.sendRequest(
+        `/sendMessage?&chat_id=${this.config.telegram_chat_id}&parse_mode=html&text=${msg}`
+      )
+    ).ok;
   }
 
-  public async createMessageBody (
+  public async createMessageBody(
+    id: number,
     dateFrom: DateTime | Date,
     dateTo: DateTime | Date,
-    apartmentId: number,
-    fixedPrice: number,
+    fixedTotalPrice: number,
     name: string,
-    email: string,
-    phone: string
-  ): Promise<boolean> {
-    const msg = `
-  from: ${dateFrom},
-  to:  ${dateTo},
-  apartmentId:  ${apartmentId},
-  fixedPrice:  ${fixedPrice},
-  name:  ${name},
-  email: ${email},
-  phone:  ${phone}
-  `
-
-    return await this.sendMsgToTelegram(msg)
+    phone: string,
+    payment_url: string
+    ): Promise<boolean> {
+    const msg =encodeURIComponent(`
```

Comment:
```suggestion
    const msg = encodeURIComponent(`
```
Этот метод для шифровки URL, а не тела.

Reply:
я исходил из этого:
https://stackoverflow.com/questions/332872/encode-url-in-javascript
![image](https://user-images.githubusercontent.com/81852394/175764147-739d299f-605c-4ffd-ab0d-5209f0b4eb2e.png)

так как этот метод наиболее всеяден, а какие символы могут быть в ссылке на оплату я на нашёл.

Reply:
Все, увидел выше, что это не тело, а query param. Посмотри, может axios это уже сам делает, тогда не нужно. Если нет, то да, нужно.